libraries/lib-wave-track/WaveClip.cpp

```diff
@@ -531,6 +529,8 @@ bool WaveClip::Paste(double t0, const WaveClip &other)
    for (size_t ii = 0, width = GetWidth(); ii < width; ++ii)
       mSequences[ii]->Paste(s0, newClip->mSequences[ii].get());
 
+   transaction.Commit();
```

Comment:
So, I should change my Transaction class to save and restore the state of trims too.  That's a good point.

Some years ago I made a very thorough examination of places where saving a new block (then, separate files, now, rows in the Sqlite database) might fail because of limited drive space.  These places can throw exceptions.

Most exceptions are handled in `AudacityApp::OnExceptionInMainLoop()` which uses the last undo/redo state to restore the state of the project in memory.  So inconsistencies of the trims would be corrected.

But there is also `AudioIO::DrainRecordBuffers()` which does its own exception handling, in case space runs out during recording.  It keeps as much recording as possible from the last successful autosave.

There are other places where `GuardedCall` stops exceptions.